File: ExpenseTracker_backend/src/main/java/com/expensetracker/service/expense/ExpenseService.java
package com.expensetracker.service.expense;


import java.util.List;

import com.expensetracker.dto.*;
import com.expensetracker.entity.*;

public interface ExpenseService {
	
	Expense postExpense(ExpenseDTO expenseDTO);
	
	List<Expense> getAllExpenses();
	
	Expense getExpenseById(Long id);
	
	Expense updateExpense(Long id,ExpenseDTO expenseDTO);
	
	void deleteExpense(Long id);
	
}

File: ExpenseTracker_frontend/expensetrackerwebapp/src/app/services/expense/expense.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';


// const baseUrl = 'http://localhost:5001/';
const baseUrl = 'http://expensetracker-env.eba-vp6hmw64.eu-north-1.elasticbeanstalk.com/';

@Injectable({
  providedIn: 'root'
})
export class ExpenseService {

  constructor(private http:HttpClient) { }

  postExpense(expenseDTO:any):Observable<any>{
    return this.http.post(baseUrl+"api/expense",expenseDTO);
  }

  getAllExpenses():Observable<any>{
    return this.http.get(baseUrl+"api/expense/all");
  }

  deleteExpense(id:number):Observable<any>{
    return this.http.delete(baseUrl+"api/expense/"+id);
    
  }

  getExpenseById(id:number):Observable<any>{
    return this.http.get(baseUrl+"api/expense/"+id);
  }

  updateExpense(id:number,expenseDTO:any):Observable<any>{
    return this.http.put(baseUrl+"api/expense/"+id,expenseDTO);
  }
}
